### PR TITLE
feat: Copy to clipboard component

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Introducing copy to clipboard as a coded component.

### How has this been tested?

New unit tests and a page for screenshot tests.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/copy-to-clipboard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        wrapText={false}
+        formAction="none"
+      />
+    </InternalPopover>
+  );
+
+  return (
+    <div ref={__internalRootRef} {...baseProps} className={clsx(baseProps.className, styles.root, testStyles.root)}>
+      {variant === 'inline' ? (
+        <div className={styles['inline-container']}>
```
Comment (just-boris) [inline src/copy-to-clipboard/index.tsx:85]: If this is inline, why are these divs? What about appropriate HTML nesting? Such as using copy-to-clipboard inside `<p>`?

File: src/copy-to-clipboard/styles.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+.inline-container {
+  display: inline-block;
+  word-break: break-all;
+
+  &-trigger {
+    display: inline-block;
+    margin-inline-end: awsui.$space-scaled-xxs;
+  }
+  &-text {
+    display: contents;
```
Comment (just-boris) [inline src/copy-to-clipboard/styles.scss:21]: How many of these styles would be needed after addressing the comment above about `<div>`?

File: src/popover/internal.tsx [hunk 3]
```diff
@@ -141,7 +143,7 @@ function InternalPopover(
       className={clsx(popoverClasses, styles['popover-content'])}
       data-awsui-referrer-id={referrerId}
     >
-      {visible && (
+      {visible && !__hidden && (
```
Comment (just-boris) [inline src/popover/internal.tsx:146]: On the API discussion, there was a question if this private API in the scope of this feature or not. Please discuss with the team before pushing it.

File: src/copy-to-clipboard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+  const triggerVariant = (
+    {
+      button: 'normal',
+      icon: 'icon',
+      inline: 'inline-icon',
+    } as const
+  )[variant];
+
+  const trigger = (
+    <InternalPopover
```
Comment (just-boris) [inline src/copy-to-clipboard/index.tsx:63]: Should we set `renderWithPortal=true` from the start to support correct positions inside table, for example?

File: src/test-utils/dom/copy-to-clipboard/index.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  findCopyButton(): ButtonWrapper {
+    return this.findComponent(`.${ButtonWrapper.rootSelector}`, ButtonWrapper)!;
+  }
+
+  findStatusText(): null | ComponentWrapper {
+    return this.findComponent(`.${PopoverWrapper.rootSelector}`, PopoverWrapper)!.findContent();
+  }
+
+  findTextToCopy(): null | ComponentWrapper {
```
Comment (just-boris) [inline src/test-utils/dom/copy-to-clipboard/index.ts:19]: Why do these return a `ComponentWrapper`?

File: src/copy-to-clipboard/index.tsx (hunk 1 above)
Comment (pan-kot) [reply src/copy-to-clipboard/index.tsx:85]: Will update

File: src/test-utils/dom/copy-to-clipboard/index.ts (hunk 5 above)
Comment (pan-kot) [reply src/test-utils/dom/copy-to-clipboard/index.ts:19]: Should be ElementWrapper, will update 👍

File: src/popover/internal.tsx (hunk 3 above)
Comment (pan-kot) [reply src/popover/internal.tsx]: Removed this for now. We can add it later if really needed.

File: src/copy-to-clipboard/styles.scss (hunk 2 above)
Comment (just-boris) [reply src/copy-to-clipboard/styles.scss:19]: @pan-kot this question remains open. Did you try cleaning up these styles?

**Review by just-boris — COMMENTED** [review]
[One comment](https://github.com/cloudscape-design/components/pull/1782#discussion_r1427863827) from the previous review left unaddressed

File: src/copy-to-clipboard/styles.scss (hunk 2 above)
Comment (pan-kot) [reply src/copy-to-clipboard/styles.scss]: Thanks, updated now

**Review by just-boris — APPROVED** [review]

**Review by just-boris — APPROVED** [review]

**Pull request merged**